node/src/reactor/joiner.rs

```diff
@@ -978,6 +1066,210 @@ impl Reactor {
         self.rest_server.finalize().await;
         Ok(config)
     }
+
+    // If we have a latest block header, we joined via the joiner reactor.
+    //
+    // If not, run genesis or upgrade and construct a switch block, and use that for the latest
+    // block header.
```

Comment:
That comment is outdated. Maybe:

```suggestion
    // If we are at genesis or after an upgrade, constructs the genesis or upgrade block.
    // Otherwise returns the latest block header.
```